src/reporter/index.js

```diff
@@ -46,31 +47,29 @@ export default class Reporter {
         return find(this.reportQueue, i => i.test === testRun.test);
     }
 
-    _shiftReportQueue () {
-        // NOTE: a completed report item is always at the front of the queue.
-        // This happens because the next test can't be completed until the
-        // previous one frees all browser connections.
-        // Therefore, tests always get completed sequentially.
-        var reportItem = this.reportQueue.shift();
+    _shiftReportQueue (reportItem) {
+        var currentFixture = null;
 
-        if (!reportItem.errs.length && !reportItem.test.skip)
-            this.passed++;
+        while (this.reportQueue.length && this.reportQueue[0].testRunInfo) {
+            reportItem     = this.reportQueue.shift();
+            currentFixture = reportItem.fixture;
 
-        this.plugin.reportTestDone(reportItem.test.name, Reporter._createTestRunInfo(reportItem));
+            this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo);
+        }
```

Comment:
It seems, if you have the following tests:
```
fixture1
  test1
fixture2
  test2
...
```
`concurrency 2` and `test2` finished before `test1` the report will be the following:
```
fixture1
  test1
  test2
fixture2